backend/ftp/dataconn.go

```diff
@@ -152,7 +147,7 @@ func getDataConn(ctx context.Context, authority utils.Authority, fs *FileSystem,
 		fs.dataconn = nil
 	}
 
-	if fs.dataconn == nil || (f != nil && f.resetConn) {
+	if fs.dataconn == nil || fs.resetConn {
```

Comment:
moved this reset flag from `f.restConn` to the `f.filesystem.resetConn` for the following scenario:
```go
_, _ = f.Write("something")
_ = f.Close() // this closes all dataconn connections and marks it as resetConn=true

otherFile := f.Location().NewFile("newfile.txt")
otherFile.Read(&buffer)
```
This would fail because Read would call `otherfile.filesystem.dataconn.R.Read()` but R is nil (thus the need for resetConn) but Location had no knowledege of the resetConn value.  So it really belongs at the filesytem level.